src/main/java/com/elasticbox/jenkins/SlaveConfiguration.java

```diff
@@ -261,25 +261,38 @@ public ListBoxModel doFillBoxVersionItems(@RelativePath("..") @QueryParameter St
             return DescriptorHelper.getBoxVersions(createClient(endpointUrl, token), workspace, box);
         }
 
-        public ListBoxModel doFillProfileItems(@RelativePath("..") @QueryParameter String endpointUrl,
+        public ListBoxModel doFillProfileItems(@RelativePath("..") @QueryParameter String name,
+                                               @RelativePath("..") @QueryParameter String endpointUrl,
                                                 @RelativePath("..") @QueryParameter String token,
                                                 @QueryParameter String workspace,
                                                 @QueryParameter String box) {
 
-            ListBoxModel profiles = getEmptyListBoxModel("--Please choose policy box--", "");
             if (anyOfThemIsBlank(endpointUrl, workspace, box, token)) {
                 return getEmptyListBoxModel();
             }
 
+            Client client = (name != null && name.length() > 0)
+                    ? ClientCache.getClient(name) : new Client(endpointUrl, token);
```

Comment:
Hmmm, yes, maybe if the client is already cached but the user has changed the URL and not saved yet, we must check the `endpointUrl`... I changed the code, but in fact it affected most of the methods, so I changed all of them